x-pack/plugins/apm/public/hooks/use_dynamic_data_view.ts

```diff
@@ -5,17 +5,23 @@
  * 2.0.
  */
 
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { ApmPluginStartDeps } from '../plugin';
 import { useFetcher } from './use_fetcher';
 
 export function useDynamicDataViewFetcher() {
-  const { data, status } = useFetcher((callApmApi) => {
-    return callApmApi('GET /internal/apm/data_view/dynamic', {
-      isCachable: true,
-    });
-  }, []);
+  const { services } = useKibana<ApmPluginStartDeps>();
 
-  return {
-    dataView: data?.dynamicDataView,
-    status,
-  };
+  const { data, status } = useFetcher(
+    async (callApmApi) => {
+      const res = await callApmApi('GET /internal/apm/data_view/title', {
+        isCachable: true,
+      });
+
+      return services.dataViews.create({ title: res.apmDataViewTitle });
```

Comment:
Previously the endpoint returned a fake data view. Now it only returns the data view index pattern (aka title) which is used to build a real (non-persisted) data view.

Reply:
It would be great if we can just have one hook / endpoint instead of having a static and a dynamic data view. Now that we use the `dataViews` services it is more straightforward to do so.